### PR TITLE
addProps()

### PR DESCRIPTION
**Pull request by peterstuart**
Working up to `addHandlers()`, which is trickier.

### CODE REVIEW

File: types/ad-hok/index.d.ts [hunk 1]
```diff
@@ -1,31 +1,25 @@
...
-    stateUpdaterName: TStateUpdaterName,
-    initialState: TState | ((props: TProps) => TState),
-  ) => (
-    props: TProps,
-  ) => TProps &
-    { [stateName in TStateName]: TState } &
-    { [stateUpdaterName in TStateUpdaterName]: (state: TState) => void }
-
-  declare const addState: AddStateType
-}
```
Comment (peterstuart) [inline types/ad-hok/index.d.ts]: Not sure what the diff is here?

Comment (helixbass) [reply types/ad-hok/index.d.ts]: Ya maybe line endings/tabs or something like that?

File: types/ad-hok/index.d.ts [hunk 2]
```diff
@@ -1,31 +1,25 @@
...
+  ) => (
+    props: TProps,
+  ) => TProps &
+    { [stateName in TStateName]: TState } &
+    { [stateUpdaterName in TStateUpdaterName]: (state: TState) => void }
+
+  declare const addState: AddStateType
+
+  type AddPropsType = <TProps, AdditionalProps extends { [key: string]: any }>(
+    createProps: (props: TProps) => AdditionalProps,
```
Comment (helixbass) [inline types/ad-hok/index.d.ts]: Should also support the object literal form of `addProps()` eg `addProps({a: 1})`

File: types/ad-hok/index.d.ts [hunk 3]
```diff
@@ -1,31 +1,25 @@
...
+    dependencies?: Array<keyof TProps>,
```
Comment (helixbass) [inline types/ad-hok/index.d.ts]: Same comment about dependencies as #4 applies here too

File: types/ad-hok/index.d.ts (hunk 2 above)
Comment (peterstuart) [reply types/ad-hok/index.d.ts]: 👍

**Pull request merged**